src/System.Net.Security/tests/FunctionalTests/SslStreamAlpnTests.cs

```diff
@@ -20,8 +20,68 @@ namespace System.Net.Security.Tests
 {
     using Configuration = System.Net.Test.Common.Configuration;
 
+#if TargetsLinux
+    internal static class OpenSslVersionProvider
+    {
+        private static string s_opensslVersion;
+
+        [DllImport("System.Security.Cryptography.Native.OpenSsl", EntryPoint = "CryptoNative_SSLEayVersion")]
+        private static extern string OpenSslVersionDescription();
+
+        private static string OpenSslVersionNumber
+        {
+            get
+            {
+                if (s_opensslVersion == null)
+                {
+                    const string OpenSSL = "OpenSSL ";
+
+                    // Skip OpenSSL part, and get the version string of format x.y.z
+                    s_opensslVersion  = OpenSslVersionDescription().Substring(OpenSSL.Length, 5);
+                }
+
+                return s_opensslVersion;
+            }
+        }
+
+        internal static int MajorVersion
+        {
+            get
+            {
+                int digit = OpenSslVersionNumber[0] - '0';
+                return (digit >= 0 && digit <= 9) ? digit : -1;
+            }
+        }
+
+        internal static int MinorVersion
+        {
+            get
+            {
+                int digit = OpenSslVersionNumber[2] - '0';
+                return (digit >= 0 && digit <= 9) ? digit : -1;
+            }
+        }
+
+        internal static int BuildVersion
+        {
+            get
+            {
+                int digit = OpenSslVersionNumber[4] - '0';
+                return (digit >= 0 && digit <= 9) ? digit : -1;
+            }
+        }
```

Comment:
Can `Version.Parse` or `Version.TryParse` be used instead of all of this?  Then a) you don't have to do the parsing yourself, and b) this can be simplified to expose a single Version-returning property.  Or the caller could even do it based on OpenSslVersionNumber.

Reply:
Yeah, I can use that, didn't know about that API.